x-pack/test/functional/services/ml/settings_calendar.ts

```diff
@@ -9,16 +9,73 @@ import expect from '@kbn/expect';
 
 import { FtrProviderContext } from '../../ftr_provider_context';
 import { MlCommonUI } from './common_ui';
+import { Navigation } from './navigation';
+import { MlADJobTable } from './job_table';
 
 export function MachineLearningSettingsCalendarProvider(
   { getService }: FtrProviderContext,
-  mlCommonUI: MlCommonUI
+  mlCommonUI: MlCommonUI,
+  navigation: Navigation,
+  jobTable: MlADJobTable
 ) {
   const testSubjects = getService('testSubjects');
   const retry = getService('retry');
   const comboBox = getService('comboBox');
 
   return {
+    async checkForCalendar(jobNumbers: number[], checkForExists: boolean = true) {
+      for await (const jobNumber of jobNumbers)
+        await jobTable.withDetailsOpen(`test_calendar_ad_${jobNumber}`, async () => {
+          await testSubjects[checkForExists ? 'existOrFail' : 'missingOrFail'](
+            jobTable.detailsSelector(
+              `test_calendar_ad_${jobNumber}`,
+              'mlJobRowDetailsSection-calendars'
+            )
+          );
+        });
+    },
```

Comment:
A few comments around this:
1. I think this should be a method of the job table service
2. In the current form, this is very specific to the way how the test file named the calendars, which makes it less reusable. I'd suggest to make it a bit more generic (as mentioned in a comment of my first review): something like `assertJobRowCalendars(jobId: string, expectedCalendars: string[])`

Reply:
Sorry about the oversight from the first review.

Reply:
Does [this](https://github.com/elastic/kibana/pull/178553/commits/b4529e5b647c5c15fa43cc68a11dc3095d748ad7) suffice?